webserver/server/app/services/TransactionHistoryService.ts

```diff
@@ -74,9 +87,29 @@ export async function historyForAddresses(
       },
 
       transaction: {
-        hash: entry.hash.toString('hex'),
-        payload: entry.payload.toString('hex'),
+        ...{
+          hash: entry.hash.toString('hex'),
+          payload: entry.payload.toString('hex'),
+        },
+        ...(request.withInputContext && {
+          metadata: entry.metadata && entry.metadata.toString('hex'),
+          inputCredentials: entry.input_addresses
+            ? (entry.input_addresses as string[]).map(getPaymentCred)
+            : [],
+        }),
       },
     })),
   };
 }
+
+function getPaymentCred(addressRaw: string): string {
+  const address = Address.from_bytes(Buffer.from(addressRaw.slice(2), 'hex'));
+
+  const paymentCred = address.payment_cred();
+  const addressBytes = paymentCred?.to_bytes();
+
+  address.free();
+  paymentCred?.free();
```

Comment:
fyi you no longer need to free manually with CML v4

Reply:
good to know, but carp is still on `3.0.1` (https://github.com/dcSpark/carp/blob/f44fd3307d5c98b4c02f81b0b190c040b202089c/webserver/server/package.json#L28)

Reply:
yeah, it will switch with this PR: https://github.com/dcSpark/carp/pull/159

Reply:
that's currently only for the indexer though, I don't see a change in the webserver version there

Reply:
we'll want to upgrade it (maybe in a separate PR)